### PR TITLE
chore: consider runsettings TestRunParameters in context/browser creation

### PR DESCRIPTION
**Pull request by mxschmitt**
Relates https://github.com/microsoft/playwright-dotnet/issues/1730 we'll follow-up with docs to close it.

### CODE REVIEW

File: src/Playwright.NUnit/PlaywrightTest.cs [hunk 1]
```diff
@@ -22,15 +22,14 @@
...
-using System;
 using System.Threading.Tasks;
 using NUnit.Framework;
 
 namespace Microsoft.Playwright.NUnit
 {
     public class PlaywrightTest : WorkerAwareTest
     {
-        public static string BrowserName => (Environment.GetEnvironmentVariable("BROWSER") ?? Microsoft.Playwright.BrowserType.Chromium).ToLower();
+        public string BrowserName { get; internal set; } = null!;
```
Comment (yury-s) [inline src/Playwright.NUnit/PlaywrightTest.cs:32]: This is a breaking change, isn't it? I believe we should support both, probably giving priority to the env variable.

File: src/Playwright.Tests/TestConstants.cs [hunk 2]
```diff
@@ -33,17 +33,36 @@ namespace Microsoft.Playwright.Tests
...
+
+        internal static bool IsChromium
+        {
+            get
+            {
+                return BrowserName == BrowserType.Chromium;
+            }
+        }
 
-        internal static readonly bool IsWebKit = BrowserName == BrowserType.Webkit;
```
Comment (yury-s) [inline src/Playwright.Tests/TestConstants.cs:44]: nit: Why did it change? The previous code was more concise.

File: src/Playwright/Core/Shared/RunSettingsParser.cs [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+                    return;
+                }
+            }
+            var type = Nullable.GetUnderlyingType(property.PropertyType) ?? property.PropertyType;
+            switch (type)
+            {
+                case Type t when t == typeof(string):
+                    property.SetValue(options, value);
+                    break;
+                case Type t when type == typeof(bool):
```
Comment (yury-s) [inline src/Playwright/Core/Shared/RunSettingsParser.cs:85]: `Type t when t ==` for consistency with other branches? I'm not actually sure what `Type t` would yield in this particular expression, is this case covered with a test?

File: src/Playwright.NUnit/PlaywrightTest.cs (hunk 1 above)
Comment (yury-s) [reply src/Playwright.NUnit/PlaywrightTest.cs:32]: Ok, I see that it moved to the settings parser.

**Review by yury-s — APPROVED** [review]

File: src/Playwright.MSTest/ContextTest.cs [hunk 4]
```diff
@@ -35,9 +35,12 @@ public class ContextTest : BrowserTest
...
         {
-            Context = await NewContextAsync(ContextOptions).ConfigureAwait(false);
+            Context = await NewContextAsync(ContextOptions()).ConfigureAwait(false);
         }
 
-        public virtual BrowserNewContextOptions? ContextOptions => null;
+        public virtual BrowserNewContextOptions ContextOptions()
+        {
+            return null!;
+        }
```
Comment (kblok) [inline src/Playwright.MSTest/ContextTest.cs:44]: Why do we need this change?

File: src/Playwright.Tests/TestConstants.cs (hunk 2 above)
Comment (mxschmitt) [reply src/Playwright.Tests/TestConstants.cs:44]: BrowserName gets set lazily, so we need a getter which gets executed.

This is only for our internal tests too.

File: src/Playwright.MSTest/ContextTest.cs (hunk 4 above)
Comment (mxschmitt) [reply src/Playwright.MSTest/ContextTest.cs:44]: Align it with nunit harness.

File: src/Playwright.Tests/TestConstants.cs [hunk 5]
```diff
@@ -33,17 +33,36 @@ namespace Microsoft.Playwright.Tests
...
     {
-        public static string BrowserName => NUnit.PlaywrightTest.BrowserName;
+        public static string BrowserName { get; set; } = null!;
 
         public const int DefaultTestTimeout = 30_000;
         public const int DefaultTimeout = 10_000;
         public const int DefaultTaskTimeout = 5_000;
 
-        public const string AboutBlank = "about:blank";
+        internal static bool IsWebKit
```
Comment (campersau) [inline src/Playwright.Tests/TestConstants.cs:42]: Nit: you can use the shorthand syntax if you want:
```suggestion
        internal static bool IsWebKit => BrowserName == BrowserType.Webkit;
```

File: src/Playwright/Core/Shared/RunSettingsParser.cs (hunk 3 above)
Comment (mxschmitt) [reply src/Playwright/Core/Shared/RunSettingsParser.cs:85]: will follow up with better tests.

File: src/Playwright.Tests/TestConstants.cs (hunk 5 above)
Comment (mxschmitt) [reply src/Playwright.Tests/TestConstants.cs:42]: sweet.

**Pull request merged**